tests/testthat/test-reporter.R

```diff
@@ -77,4 +82,111 @@ test_that("reporters produce consistent output", {
   save_report("silent")
   save_report("rstudio")
   save_report("junit", reporter = createJunitReporterMock())
+
+  # Test that MultiReporter can write to two different places
+  tap_file <- tempfile()
+  save_report("summary", reporter = MultiReporter$new(list(
+    SummaryReporter$new(show_praise = FALSE, omit_dots = FALSE),
+    TapReporter$new(file=tap_file))
+  ))
+  expect_identical(read_lines(tap_file),
+    read_lines(test_path("reporters", "tap.txt")))
+})
+
+
+test_report_to_file <- function(name, reporter = find_reporter_one(name, ...),
+                                output_file, error_regexp=NA, ...) {
+  # output_file is where we expect output to be written to, whether we pass it
+  # as an argument to Reporter$new() (here, via the ...), or whether it is set
+  # in an option.
+  path <- test_path("reporters", paste0(name, ".txt"))
+  expect_silent(
+    expect_error(
+      test_reporter(reporter),
+      error_regexp
+    )
+  )
+  expect_identical(read_lines(output_file), read_lines(path))
+}
+
+test_that("reporters accept a 'file' arugment and write to that location", {
+  output <- tempfile()
+  test_report_to_file("check", file = output, error_regexp=NULL, output_file = output)
+  test_report_to_file(
+    "progress",
+    ProgressReporter$new(show_praise = FALSE, min_time = Inf, file = output),
+    output_file = output
+  )
+  test_report_to_file(
+    "summary",
+    SummaryReporter$new(show_praise = FALSE, omit_dots = FALSE, file = output),
+    output_file = output
+  )
+  test_report_to_file(
+    "summary-2",
+    SummaryReporter$new(show_praise = FALSE, max_reports = 2, file = output),
+    output_file = output
+  )
+  test_report_to_file(
+    "summary-no-dots",
+    SummaryReporter$new(show_praise = FALSE, omit_dots = TRUE, file = output),
+    output_file = output
+  )
+  test_report_to_file("location", file = output, output_file = output)
+  test_report_to_file("minimal", file = output, output_file = output)
+  test_report_to_file("tap", file = output, output_file = output)
+  test_report_to_file("teamcity", file = output, output_file = output)
+  test_report_to_file("rstudio", file = output, output_file = output)
+  test_report_to_file(
+    "junit",
+    reporter = createJunitReporterMock(file = output),
+    output_file = output
+  )
+})
+
+test_that("reporters write to 'testthat.output_file', if specified", {
```

Comment:
I think this only needs one test since it's only one line of code

Reply:
It was in testing all of the reporters that I found that several of them needed `...` added to their `initialize` methods to pass the `file` argument through to `super$initialize`, so there is actually more than just the one line of code exercised by these tests.

Reply:
I think that applies to the test case above, not this one?  (Because the option is set in the base class initialiser)

Reply:
Ah sorry, misread. You're right, will prune.